pkg/util/system/namespace_linux.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package system
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"syscall"
+)
+
+var (
+	netNSPid1     uint64
+	syncNetNSPid1 sync.Once
+)
+
+// GetProcessNamespaceInode performs a stat() call on /proc/<pid>/ns/<namespace>
+func GetProcessNamespaceInode(procPath string, pid int, namespace string) (uint64, error) {
+	nsPath := filepath.Join(procPath, strconv.Itoa(pid), "ns", namespace)
+	fi, err := os.Stat(nsPath)
+	if err != nil {
+		return 0, err
+	}
+
+	// We are on linux, casting in safe
+	return fi.Sys().(*syscall.Stat_t).Ino, nil
+}
+
+func IsProcessHotNetwork(procPath string, namespaceID uint64) *bool {
```

Comment:
Needs a docstring.  I don't know what a "hot" network is?

And tests :)

Reply:
Also fixed